Add Fleet & Agent 8.16.0 Release Notes

kilfoyle opened the pull request:
This adds the 8.16.0 Fleet & Elastic Agent Release Notes:

* Fleet contents from [Kibana Release Notes PR](https://github.com/elastic/kibana/pull/198166)
* Fleet Server contents from [BC3 changelog](https://github.com/elastic/fleet-server/tree/7f12dbd1abf6ceed06fbe4eb96fe1cc3e1b20219/changelog/fragments)
* Elastic Agent contents from [BC3 changelog](https://github.com/elastic/elastic-agent/tree/6c0d2398399dfbf6a4f5d95e3e55a138fee9a737/changelog/fragments)

See [DOCS PREVIEW](https://ingest-docs_bk_1412.docs-preview.app.elstc.co/guide/en/fleet/master/release-notes-8.16.0.html)

Closes: https://github.com/elastic/ingest-docs/issues/1411

cmacknz commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,237 @@
+* Advanced agent monitoring options UI for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]
+* Add option to have Kafka dynamic topics in outputs. {kibana-pull}192720[#192720]
+* Add support for GeoIP processor databases in Ingest Pipelines. {kibana-pull}190830[#190830]
+* Add agentless UX creation flow. {kibana-pull}189932[#189932]
```
@kpollich do we want anything to do with agentless in the release notes yet, considering it isn't usable or really announced yet?

cmacknz commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,237 @@
+* Enable feature flag for reusable integration policies. {kibana-pull}187153[#187153]
+
+{fleet-server}::
+* Add `/api/fleet/agents/:id/audit/unenroll` API that an {agent} or Endpoint process may use to enable accurate status reporting in the {fleet} UI. {fleet-server-pull}3818[#3818] {agent-issue}484[#484]
```
```suggestion
* Add `/api/fleet/agents/:id/audit/unenroll` API that an {agent} or Endpoint process may use to report that an agent was uninstalled or unenrolled to {fleet}. {fleet-server-pull}3818[#3818] {agent-issue}484[#484]
```

cmacknz commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,237 @@
+
+{agent}::
+* Uninstalling a {fleet}-managed {agent} instance will now do a best-effort attempt to notify {fleet-server} of the agent removal so the agent status appears correctly in the {fleet} UI (related to {fleet-server-pull}3818[#3818] above). {agent-pull}5302[#5302] {agent-issue}484[#484]
+* Remove support for the experimental shippers feature. Removal is non-breaking since the feature was experimental only. {agent-pull}5308[#5308] {agent-issue}4547[#4547]
```
```suggestion
* Remove support for the experimental shippers feature. {agent-pull}5308[#5308] {agent-issue}4547[#4547]
```

kilfoyle commented:
@cmacknz As you suggested in the feature summary issue, I've added in descriptions from that issue for the release features, in commit [9323922](https://github.com/elastic/ingest-docs/pull/1412/commits/9323922dc3ecdb2797281d488c53100e027b47e1).

The "installing agent using Helm" feature wasn't in the release notes originally, I think because [the PR](https://github.com/elastic/elastic-agent/pull/5331) has a `skip-changelog` tag. I've added that PR into the features section but let me know if it shouldn't be there for any reason.

cmacknz commented:
The Helm chart being GA should definitely be in here, thanks

kpollich commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,239 @@
+
+{fleet}::
+* Support content packages in UI. {kibana-pull}195831[#195831]
+* Advanced agent monitoring options UI for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]
```
```suggestion
* Add advanced agent monitoring options for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]
```

kpollich commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,239 @@
+* Add option to have Kafka dynamic topics in outputs. This allows the Kafka output to write to a topic which is dynamically set in an event field. {kibana-pull}192720[#192720]
+* Add support for GeoIP processor databases in Ingest Pipelines. {kibana-pull}190830[#190830]
+* Add agentless UX creation flow. {kibana-pull}189932[#189932]
+* Enable feature flag for reusable integration policies. This feature allows you to create integrations policies that can be shared with multiple {agent} policies, thereby reducing the number of integrations policies that you need to actively manage. {kibana-pull}187153[#187153]
```
```suggestion
* Add support for reusable/shareable integration policies. This feature allows you to create integrations policies that can be shared with multiple {agent} policies, thereby reducing the number of integrations policies that you need to actively manage. {kibana-pull}187153[#187153]
```

kpollich commented on docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc:
```diff
@@ -0,0 +1,239 @@
+* Add support for GeoIP processor databases in Ingest Pipelines. {kibana-pull}190830[#190830]
+* Add agentless UX creation flow. {kibana-pull}189932[#189932]
+* Enable feature flag for reusable integration policies. This feature allows you to create integrations policies that can be shared with multiple {agent} policies, thereby reducing the number of integrations policies that you need to actively manage. {kibana-pull}187153[#187153]
+* Support integration-level outputs. This feature enables you to send integration data to a specific output, over-writing the output defined in the {agent} Policy. {kibana-pull}189125[#189125]
```
```suggestion
* Add support for integration-level outputs. This feature enables you to send integration data to a specific output, overwriting the output defined in the {agent} Policy. {kibana-pull}189125[#189125]
```

kilfoyle commented:
Thanks @kpollich. I've added in all of your suggestions.

kilfoyle commented:
I've added a breaking change for Elastic Agent based (Beats [39993](https://github.com/elastic/beats/issues/39993)) as suggested by Luca [here](https://github.com/elastic/ingest-dev/issues/4013#issuecomment-2459754344).

kpollich reviewed (approved):
LGTM 🚀
